app/src/main/java/com/example/happybirthday/unit3/LetterAdapter.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.happybirthday.unit3
+
+import android.content.Intent
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.accessibility.AccessibilityNodeInfo
+import android.widget.Button
+import androidx.annotation.RequiresApi
+import androidx.navigation.findNavController
+import androidx.recyclerview.widget.RecyclerView
+import com.example.happybirthday.R
+
+/**
+ * Adapter for the [RecyclerView] in [MainActivity].
+ */
+class LetterAdapter :
+    RecyclerView.Adapter<LetterAdapter.LetterViewHolder>() {
+
+    // Generates a [CharRange] from 'A' to 'Z' and converts it to a list
+    private val list = ('A').rangeTo('Z').toList()
+
+    /**
+     * Provides a reference for the views needed to display items in your list.
+     */
+    class LetterViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
+        val button = view.findViewById<Button>(R.id.button_item)
+    }
+
+    override fun getItemCount(): Int {
+        return list.size
+    }
+
+    /**
+     * Creates new views with R.layout.item_view as its template
+     */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LetterViewHolder {
+        val layout = LayoutInflater
+            .from(parent.context)
+            .inflate(R.layout.item_view, parent, false)
+        // Setup custom accessibility delegate to set the text read
+        layout.accessibilityDelegate = Accessibility
```

Comment:
in general it's good to remember about accessibility. But in real life it's usually omitted. So if something is not clear about it or you need to spend extra time, you can skip it, it's not very important for us now